File: vigil-backend/app/schemas/commit.py
from datetime import datetime
from typing import List, Optional
import uuid
from pydantic import BaseModel, ConfigDict
from app.schemas.base import PaginatedResponse


class CommitRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    repository_id: uuid.UUID
    sha: str
    message: str
    author_login: Optional[str] = None
    author_name: Optional[str] = None
    author_email: Optional[str] = None
    committed_at: Optional[datetime] = None
    parent_sha: Optional[str] = None
    created_at: datetime


class CommitListResponse(PaginatedResponse):
    items: List[CommitRead]

File: vigil-frontend/src/types/commit.ts
import type { PaginatedResponse } from './api';

export interface CommitRead {
  id: string; // uuid
  repository_id: string; // uuid
  sha: string;
  message: string;
  author_login: string | null;
  author_name: string | null;
  author_email: string | null;
  committed_at: string | null;
  parent_sha: string | null;
  created_at: string;
}

export type CommitListResponse = PaginatedResponse<CommitRead>;
